File: Week 1&2 - Internet Foundations/Python/BenchMark.py
import random
import time
from MergeSort import merge_sort

numbers = [random.randint(1, 10000) for _ in range(1000)]

start = time.time()
sorted_numbers = merge_sort(numbers)
end = time.time()

print(f"Runtime: {end - start:.6f} seconds")

File: Week 1&2 - Internet Foundations/NodeJS/BenchMark.js
const mergeSort = require("./mergeSort");

function generateRandomNumbers(count, min, max) {
  return Array.from({ length: count }, () =>
    Math.floor(Math.random() * (max - min + 1)) + min
  );
}

const numbers = generateRandomNumbers(1000, 1, 10000);

const start = performance.now();

mergeSort(numbers);

const end = performance.now();

console.log(`Runtime: ${(end - start).toFixed(4)} milliseconds`);